### ISSUE TITLE
Visualizing Latent points before decoding?

### ISSUE THREAD

**Issue by Javiergonb** [nv-tlabs/LION#76]
Hi again @ZENGXH,

Is there a way to visualize the z_local tensor that is created when running the DDPM on the encoded h0?
When using the sample method of the LION model the output dict returns a z_local with (num_samples, 8192,1,1) dimensions. From my understanding this is the z_local after decoding? I am not sure how visualize in 3D the z_local that is created before the decoding, like how the show here:

![image](https://github.com/user-attachments/assets/3787fb6f-106c-401f-9959-b14e83774f48)

How did you do it?

Thanks in advance!

**Comment by ZENGXH** [comment]
could you try to reshape it to z_reshape in (num_samples, 2048, 4,1,1) and then visualize `z_reshape[:,:,:3]`? Let me know if it does not work

**Comment by Javiergonb** [comment]
Yes this works. I was trying to do something similar with the VAE. I am getting h0 from the recont method in the VAE model. This part of the code:

https://github.com/nv-tlabs/LION/blob/333c5b4903098418a2aae897ff079bcf0adc3eae/models/vae_adain.py#L202C16-L202C99

I believe this is getting a 3d representation of h0 no?
